optimum/habana/diffusers/pipelines/instantid/pipeline_stable_diffusion_xl_instantid.py

```diff
@@ -0,0 +1,1137 @@
+                negative_add_time_ids = add_time_ids
+
+            prompt_embeds = prompt_embeds.to(device)
+            if negative_prompt_embeds is not None:
+                negative_prompt_embeds = negative_prompt_embeds.to(device)
+            add_text_embeds = add_text_embeds.to(device)
+            if negative_pooled_prompt_embeds is not None:
+                negative_pooled_prompt_embeds = negative_pooled_prompt_embeds.to(device)
+            add_time_ids = add_time_ids.to(device).repeat(num_prompts * num_images_per_prompt, 1)
+            negative_add_time_ids = negative_add_time_ids.to(device).repeat(num_prompts * num_images_per_prompt, 1)
+
+            # 7.3 Split into batches (HPU-specific step)
+            (
+                latents_batches,
+                prompt_embeds_batches,
+                add_text_embeddings_batches,
+                add_time_ids_batches,
+                num_dummy_samples,
+            ) = GaudiStableDiffusionXLPipeline._split_inputs_into_batches(
+                batch_size,
+                latents,
+                prompt_embeds,
+                negative_prompt_embeds,
+                add_text_embeds,
+                negative_pooled_prompt_embeds,
+                add_time_ids,
+                negative_add_time_ids,
+            )
+
+            outputs = {
+                "images": [],
+            }
+            t0 = time.time()
+            t1 = t0
+
+            self._num_timesteps = len(timesteps)
+
+            hb_profiler = HabanaProfile(
+                warmup=profiling_warmup_steps,
+                active=profiling_steps,
+                record_shapes=False,
+            )
+            hb_profiler.start()
+
+            # 8. Denoising loop
+            throughput_warmup_steps = kwargs.get("throughput_warmup_steps", 3)
+            for j in self.progress_bar(range(num_batches)):
+                # The throughput is calculated from the 3rd iteration
+                # because compilation occurs in the first two iterations
+                if j == throughput_warmup_steps:
+                    t1 = time.time()
+
+                latents_batch = latents_batches[0]
+                latents_batches = torch.roll(latents_batches, shifts=-1, dims=0)
+                prompt_embeds_batch = prompt_embeds_batches[0]
+                prompt_embeds_batches = torch.roll(prompt_embeds_batches, shifts=-1, dims=0)
+                add_text_embeddings_batch = add_text_embeddings_batches[0]
+                add_text_embeddings_batches = torch.roll(add_text_embeddings_batches, shifts=-1, dims=0)
+                add_time_ids_batch = add_time_ids_batches[0]
+                add_time_ids_batches = torch.roll(add_time_ids_batches, shifts=-1, dims=0)
+                encoder_hidden_states = torch.cat([prompt_embeds_batch, prompt_image_emb], dim=1)
+                num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+
+                for i in range(num_inference_steps):
```

Comment:
Please change `range(num_inference_steps)`  -> `range(len(timesteps))`

to cover some schedulers which could give num of timesteps != num_inference_steps

Same fix as here:
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L515

Reply:
Thanks! done